ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -249,6 +418,11 @@ def rename_file_group(self, old, new):
             raise FileNotFoundError("No such fileGrp '%s'" % old)
         el_fileGrp.set('USE', new)
 
+        # Rename the fileGrp in both caches
+        if self._cache_flag:
+            self._fileGrp_cache[new] = self._fileGrp_cache.pop(old)
+            self._file_cache[new] = deepcopy(self._file_cache.pop(old))
```

Comment:
Why `deepcopy`?

Reply:
For a similar reason as above. To take care of the nested dictionaries properly. If `deep` is not required and simply passing the `self._file_cache[new] = self._file_cache.pop(old)` is smart enough to deal with nested data structures, it could be removed.

Reply:
Yes, then `deepcopy` is not needed here, python will reuse the existing reference just fine and do the cleanup of `pop`.